### ISSUE TITLE
CentOS 7 with netdata v1.22.1 with SSL + firefox => dashboard loading problem.

### ISSUE THREAD

**Issue by llevet** [netdata/netdata#9154]
netdata v1.22.1
Firefox 76.0.1 (64 bits) client.
Server CentOS Linux release 7.8.2003 (Core)
openssl-1.0.2k

Problem :
When loading dashborad in https only a part is loaded with firefox.
error appear in firefox console :

ChunkLoadError: "Loading chunk 0 failed.
(error: https://www.xxxxxxxxx.net:19999/static/js/0.0a2b85cc.chunk.js)"
    e (index):16
    me index.tsx:11
    Pu 6.e6ac6641.chunk.js:2
    Pu 6.e6ac6641.chunk.js:2
    Ns 6.e6ac6641.chunk.js:2
    js 6.e6ac6641.chunk.js:2
    xs 6.e6ac6641.chunk.js:2
    ys 6.e6ac6641.chunk.js:2
    oc 6.e6ac6641.chunk.js:2
    lc 6.e6ac6641.chunk.js:2
    Es 6.e6ac6641.chunk.js:2
    lc 6.e6ac6641.chunk.js:2
    render 6.e6ac6641.chunk.js:2
    188 index.tsx:15
    p (index):16
    149 main.76cfda63.chunk.js:1
    p (index):16
    i (index):16
    e (index):16
    <anonymous> main.76cfda63.chunk.js:1
Error loding source : Error: The operation was aborted. 
URL of the resource : https://www.xxxxxxxxx.net:19999/static/js/6.e6ac6641.chunk.js

Multiple refresh with F5 or shift+Refresh in firefox make it work time to time.

Trying with v1.22.1-113-nightly same problem.

Removing SSL and all work again on firefox.

The problem doesn't exist with Chrome.

*****************************************************************
Compilation ok.

[root@ns313537 netdata]# ldd /usr/sbin/netdata
        linux-vdso.so.1 =>  (0x00007ffe975cd000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f2d9f08b000)
        libz.so.1 => /lib64/libz.so.1 (0x00007f2d9ee75000)
        libcrypto.so.10 => /lib64/libcrypto.so.10 (0x00007f2d9ea12000)
        libssl.so.10 => /lib64/libssl.so.10 (0x00007f2d9e7a0000)
        libuuid.so.1 => /lib64/libuuid.so.1 (0x00007f2d9e59b000)
        libuv.so.1 => /lib64/libuv.so.1 (0x00007f2d9e36c000)
        liblz4.so.1 => /lib64/liblz4.so.1 (0x00007f2d9e157000)
        libJudy.so.1 => /lib64/libJudy.so.1 (0x00007f2d9df07000)
        libjson-c.so.2 => /lib64/libjson-c.so.2 (0x00007f2d9dcfc000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f2d9daf8000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f2d9d8dc000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f2d9d50e000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f2d9f38d000)
        libgssapi_krb5.so.2 => /lib64/libgssapi_krb5.so.2 (0x00007f2d9d2c1000)
        libkrb5.so.3 => /lib64/libkrb5.so.3 (0x00007f2d9cfd8000)
        libcom_err.so.2 => /lib64/libcom_err.so.2 (0x00007f2d9cdd4000)
        libk5crypto.so.3 => /lib64/libk5crypto.so.3 (0x00007f2d9cba1000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f2d9c999000)
        libnsl.so.1 => /lib64/libnsl.so.1 (0x00007f2d9c77f000)
        libkrb5support.so.0 => /lib64/libkrb5support.so.0 (0x00007f2d9c56f000)
        libkeyutils.so.1 => /lib64/libkeyutils.so.1 (0x00007f2d9c36b000)
        libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f2d9c151000)
        libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f2d9bf2a000)
        libpcre.so.1 => /lib64/libpcre.so.1 (0x00007f2d9bcc8000)

SSL config : 
[web]
        ssl key = /etc/netdata/ssl/privkey.pem
        ssl certificate = /etc/netdata/ssl/cert.pem

**Comment by thiagoftsm** [comment]
@jacekkolasa is the dashboard recognizing the protocol specified every time the page is loaded?

**Comment by jacekkolasa** [comment]
I don't understand how dashboard could have something to do with protocol, but maybe i'm missing something. @llevet can you access that file from last line directly? (from new tab):
> URL of the resource : xxxxxxxxx.net:19999/static/js/6.e6ac6641.chunk.js

This file is from last build, so it's not a cache problem. @thiagoftsm Do you think it could be some Agent server issue? From the error log i see that the assets are not loading sometimes

**Comment by jacekkolasa** [comment]
I was able to recreate the problem, and it always happens when i see `net::ERR_HTTP2_PROTOCOL_ERROR` response status. Looks like we'll need to experiment a little bit, there could be tons of different reasons: https://stackoverflow.com/questions/58215104/whats-the-neterr-http2-protocol-error-about

**Comment by llevet** [comment]
Attached file.
[js1.4b7938c8.chunk.js.txt](https://github.com/netdata/netdata/files/4684910/js1.4b7938c8.chunk.js.txt)

Also i have in debuger part of firefox : 
blob:https://xxxxxxxxx.net:19999/1a144a9f-407f-4ca8-af4a-6b46b6966237
Error loading the URI : Unknown source
blob:https://xxxxxxxxx.net:19999/f1871a38-8986-4e12-b7a1-750d3725be6c
Error loading the URI : Unknown source

If i copy / paste un the browser the link i have : 
File does not exist, or is not accessible: /usr/share/netdata/web/1a144a9f-407f-4ca8-af4a-6b46b6966237

The link is changin each time i relod the page.

**Comment by llevet** [comment]
Hum .... disabling gzip (enable gzip compression = no) under SSL make Chrome to not load dashboard with message in console : net::ERR_CONTENT_LENGTH_MISMATCH 200 (OK)

**Comment by thiagoftsm** [comment]
Hi @llevet ,

We brought a fix to support the newest TLS version, are you able to access Netdata without problems now?

Best regards!

**Comment by llevet** [comment]
Hi,

No i'm in version v1.23.2-51-gf32713e and the problem is still here.
One Firefox, interface is not loading (only backgroud).
On chrome, need multiple refresh F5 to work.

**Comment by thiagoftsm** [comment]
Hi @llevet ,

I executed two tests now that can help us to understand better the problem. Firstly I had the following environment:

-  Virtual Machine using CentOS 7.8.2003, Kernel 3.10.0-1127.18.2-el7 and OpenSSL 1.0.2k
-  Browser Firefox 68.11-0-1.el7

All the packages were installed today, I used `yum`. On this scenario I started the X graph using `startx` and I did not have any problem to access the content as you can see in the next image:

![ok](https://user-images.githubusercontent.com/49162938/89743990-fa270700-da97-11ea-908f-ea2650fd0ccf.png)

After to detect that with default packages on CentOS 7.x everything was working, I decided to use Slackware Current and Firefox  78.1 ESR to access the VM, and the problems started to happen. The first conclusion that I arrived is that the newest browsers are not allowing us to access the TLS version accepted by  OpenSSL 1.0.2k. I confirmed this using the next command:

```
openssl s_client -connect localhost.localdomain:19999 
```

that gave me as output:

```
New, TLSv1/SSLv3, Cipher is AES256-GCM-SHA384
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : AES256-GCM-SHA384
```

As you can see I received as output the version `TLSv1.2`. Considering that Netdata always compile using the maximum version available, this means that `TLSv1.3` is not supported on CentOS 7.x .

I have to write here that I did not have the error `ERR_CONTENT_LENGTH_MISMATCH ` on my tests, probably because the browsers could not move forward thanks the incompatibility.

> . disabling gzip (enable gzip compression = no) under SSL make Chrome to not load dashboard with message in console : net::ERR_CONTENT_LENGTH_MISMATCH 200 (OK)

This is expected, because your Netdata was compiled with support to zlib:

> libz.so.1 => /lib64/libz.so.1 (0x00007f2d9ee75000)

And as you can see in the next link

https://github.com/netdata/netdata/blob/4a4696e091cdc4deb8f482b3f837153a9c666d33/web/server/web_client.c#L1722

when this happens, Netdata will always compress the response.

Finally, I suspect that if you access your Netdata using a browser that has TLSv1.2 as maximum version, everything will work.

Best regards!

**Comment by llevet** [comment]
Hi thiagoftsm,

I access to Netdata from my desktop windows 10 with Firefox or Chrome latest version.

Trying to access Netdata in local is not relevant to real normal operation. It use loopback interface who use non standard MTU and haven't problem of packet order.

The negotiation of SSL /TLS version is between the server and the client at the beginning of the connexion. So, the problem is clearly on Netdata web server part. 
There is no problem to compile Netdata on Centos 7 wit openssl library. It will use the specific limitation of OpenSSL on Centos.
It is working with Chrome windows 10 client and Centos 7 server (must make F5 refresh time to time). The communication is well in TLS1.2, but not with Firefox.

Netdata can't negotiate TLS1.3 with client with a library who only support TLS1.2.
You also have a option on Netdata  (tls version =) to specify TLS version you use, it is TLS1.2 on my side, but no change. It suppose to give the directive on Netdata for communication.

For GZIP, a other problem of Webserver of Netdata. There is a option to disable it, but it seem to not really disable it. Only flag on http response seem to change. And it is also a part of negotiation in http protocol.

So no, disable gzip in Netdata must not create a error, it must respect its own "enable gzip compression =" parameter.

Thanks.

**Comment by thiagoftsm** [comment]
Hi @llevet ,

> Trying to access Netdata in local is not relevant to real normal operation.

I agree with you, I tested on CentOS using its default browser, to verify that the problem was happening independent of specific configuration, after to have success with this approach, I started to consider that TLS version was the main reason for we have problems. I did not have the same error than you, but I also had other TLS/SSL errors.

> Netdata can't negotiate TLS1.3 with client with a library who only support TLS1.2.

Probably if the browser is configured to access TLS 1.2, it is possible to access Netdata, because I could access the dashboard using `curl -k http://CENTOS_IP:19999`.

Best regards!
